Userland/Libraries/LibJS/Runtime/Completion.h

```diff
@@ -30,6 +31,25 @@ namespace JS {
         _temporary_result.release_value();                                                           \
     })
 
+#define MUST_OR_THROW_OOM(expression)                                                                \
+    ({                                                                                               \
+        /* Ignore -Wshadow to allow nesting the macro. */                                            \
+        AK_IGNORE_DIAGNOSTIC("-Wshadow",                                                             \
+            auto _temporary_result = (expression));                                                  \
+        if (_temporary_result.is_error()) {                                                          \
+            auto _completion = _temporary_result.release_error();                                    \
+                                                                                                     \
+            VERIFY(_completion.value().has_value());                                                 \
+            VERIFY(_completion.value()->is_object());                                                \
+            VERIFY(is<JS::InternalError>(_completion.value()->as_object()));                         \
```

Comment:
Might be good to
- Verify that `InternalError` is indeed only used for OOM errors (and "not implemented", but whatever)
- Leave a note here stating that so people understand why an OOM-allowing macro is not specifically checking for OOM